spec/filters/csv_spec.rb

```diff
@@ -197,4 +196,36 @@
       end
     end
   end
+
+  describe "using field convertion" do
+
+    let(:config) do
+      { "convert" => { "column1" => "integer", "column3" => "boolean" } }
+    end
+    let(:doc)   { "1234,bird,false" }
+    let(:event) { LogStash::Event.new("message" => doc) }
+
+    it "get converted values to the expected type" do
+      plugin.filter(event)
+      expect(event["column1"]).to eq(1234)
+      expect(event["column2"]).to eq("bird")
+      expect(event["column3"]).to eq(false)
+    end
+
+    context "when using column names" do
+
+      let(:config) do
+        { "convert" => { "custom1" => "integer", "custom3" => "boolean" },
+          "columns" => ["custom1", "custom2", "custom3"] }
+      end
+
+      it "get converted values to the expected type" do
+        plugin.filter(event)
+        expect(event["custom1"]).to eq(1234)
+        expect(event["custom2"]).to eq("bird")
+        expect(event["custom3"]).to eq(false)
+      end
+    end
+
```

Comment:
Consider adding a test case for misspelled or missing transformations, e.g. flot or upcase